scripts/nb-tester/test-notebook.py

```diff
@@ -49,16 +84,23 @@ def execute_notebook(path: Path, options: ExecuteOptions) -> bool:
         nbclient.exceptions.CellTimeoutError,
     )
     try:
-        _execute_notebook(path, options)
+        nb = _execute_notebook(path, options)
     except possible_exceptions as err:
         print("\r❌\n")
         print(err)
         return False
+
+    notebook_warnings = extract_warnings(nb)
+    if notebook_warnings:
+        print("\r⚠️")
+        [w.report() for w in notebook_warnings]
+        return True
```

Comment:
Should this return `False`? We want to error if there were warnings imo.

Reply:
I set it as `True` because some warnings are out of our control, e.g. if a Qiskit function uses a deprecated numpy function.

It depends if you'd rather have false positives or false negatives.

Reply:
What if we require you to allow-list deprecations? I'm concerned if we don't error, then no one will ever notice the warnings. People don't check CI logs proactively.

Reply:
Good point, maybe I'm over-planning; I'll set it to `False` and we can revisit it if we need.

Reply:
> because some warnings are out of our control, e.g. if a Qiskit function uses a deprecated numpy function.

Ideally that will prompt us to go fix Qiskit itself. In the meantime to them fixing it, we could allow-list it.